tests/lib/rules/no-extra-parens.js

```diff
@@ -3192,6 +3228,137 @@ ruleTester.run("no-extra-parens", rule, {
+                }
+            `,
+            output: `
+                if (condition) {
+                    /** @type {ServerOptions} */
+                    options.server.options.requestCert = false;
+                }
+            `,
+            options: ["all"],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [{ messageId: "unexpected" }]
+        },
+        {
+            code: "const net = ipaddr.parseCIDR(/** @type {string} */ (cidr));",
+            output: "const net = ipaddr.parseCIDR(/** @type {string} */ cidr);",
+            options: ["all"],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [{ messageId: "unexpected" }]
+        },
+        {
+            code: "const span = /**@type {HTMLSpanElement}*/(event.currentTarget);",
+            output: "const span = /**@type {HTMLSpanElement}*/event.currentTarget;",
+            options: ["all", { allowParensAfterCommentPattern: "invalid" }],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [{ messageId: "unexpected" }]
+        },
+        {
+            code: "if (/** @type {Compiler | MultiCompiler} */(options).hooks) console.log('good');",
+            output: "if (/** @type {Compiler | MultiCompiler} */options.hooks) console.log('good');",
+            options: ["all", { allowParensAfterCommentPattern: "invalid" }],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [{ messageId: "unexpected" }]
+        },
+        {
+            code: `
+                validate(/** @type {Schema} */ (schema), options, {
+                    name: "Dev Server",
+                    baseDataPath: "options",
+                });
+            `,
+            output: `
+                validate(/** @type {Schema} */ schema, options, {
+                    name: "Dev Server",
+                    baseDataPath: "options",
+                });
+            `,
+            options: ["all", { allowParensAfterCommentPattern: "invalid" }],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [{ messageId: "unexpected" }]
+        },
+        {
+            code: `
+                if (condition) {
+                    /** @type {ServerOptions} */
+                    (options.server.options).requestCert = false;
+                }
+            `,
+            output: `
+                if (condition) {
+                    /** @type {ServerOptions} */
+                    options.server.options.requestCert = false;
+                }
+            `,
+            options: ["all", { allowParensAfterCommentPattern: "invalid" }],
+            parserOptions: { ecmaVersion: 2020 },
+            errors: [{ messageId: "unexpected" }]
+        },
+        {
+            code: `
+                if (condition) {
+                    /** @type {ServerOptions} */
+                    /** extra coment */
+                    (options.server.options).requestCert = false;
```

Comment:
Can you also add a test with tokens between `@type` comment and parenthesized node?

Reply:
Added in https://github.com/eslint/eslint/pull/16561/commits/e29b72714c86a37bdbfe73eb7efbb92f13f1f278